integration_tests/__init__.py

```diff
@@ -129,6 +132,32 @@ def run_snapcraft_parser(self, arguments):
             raise
         return snapcraft_output
 
+    @contextmanager
+    def setup_multi_arch_sources(self, arch):
```

Comment:
This is nice, but in tests we usually have fixtures instead of context managers. I personally would prefer to follow that practice, but I don't have a very strong opinion against this context manager.